### PR TITLE
make checks work incluster

### PR DESCRIPTION
**Pull request by yuval-k**
BOT NOTES: 
resolves https://github.com/solo-io/gloo-mesh-enterprise/issues/839

### CODE REVIEW

File: pkg/meshctl/checks/check_ctx.go [hunk 1]
```diff
@@ -0,0 +1,175 @@
...
+				Namespace:   ns,
+				InCluster:   false,
+			}},
+	}
+
+}
+
+func (c *InClusterCheckContext) AccessMgmtServerAdminPort(ctx context.Context, op func(ctx context.Context, addr string) (error, string)) (error, string) {
+
+	// note: the metrics port is not exposed on the service (as it should be).
```
Comment (EItanya) [inline pkg/meshctl/checks/check_ctx.go:92]: Agreed, this will also be important for the prom operator which Amex is using

**Comment by harveyxia** [comment]
I've created an issue to track that here, https://github.com/solo-io/gloo-mesh-enterprise/issues/839, could you link this with a changelog?

File: pkg/meshctl/checks/check_ctx.go (hunk 1 above)
Comment (yuval-k) [reply pkg/meshctl/checks/check_ctx.go]: what do you mean about the prom operator?

**Review by EItanya — COMMENTED** [review]
Overall this lgtm, but I don't have that much context on this code, so I don't wanna rush to an approval

File: pkg/meshctl/checks/deployments_check.go [hunk 2]
```diff
@@ -22,47 +22,48 @@ func (d *deploymentsCheck) GetDescription() string {
...
+	installNamespace := env.Namespace
+	failure := new(Failure)
 	if len(deployments.Items) < 1 {
-		return &Failure{
-			Errors: []error{eris.Errorf("no deployments found in namespace %s", installNamespace)},
-			Hint: fmt.Sprintf(
+		failure.AddError(eris.Errorf("no deployments found in namespace %s", installNamespace))
+		if !env.InCluster {
+			failure.AddHint(fmt.Sprintf(
 				`Gloo Mesh'd installation namespace can be supplied to this cmd with the "--namespace" flag, which defaults to %s`,
```
Comment (harveyxia) [inline pkg/meshctl/checks/deployments_check.go:52]:
```suggestion
				`Gloo Mesh installation namespace can be supplied to this cmd with the "--namespace" flag, which defaults to %s`,
```

File: pkg/meshctl/checks/enterprise_registration_check.go [hunk 3]
```diff
@@ -206,39 +167,40 @@ func calculateClusterStatuses(
...
-		d.mgmtKubeConfig,
-		d.mgmtKubeContext,
-		mgmtDeployName,
-		mgmtDeployNamespace,
-		fmt.Sprintf("%v", d.localPort),
-		fmt.Sprintf("%v", d.remotePort),
-	)
+func (d *enterpriseRegistrationCheck) getConnectedAgents(ctx context.Context, checkCtx CheckContext) (map[string]int, map[string]int, error, string) {
+
+	shouldRunCheck, err := isEnterpriseVersion(ctx, checkCtx.Client(), checkCtx.Environment().Namespace)
```
Comment (harveyxia) [inline pkg/meshctl/checks/enterprise_registration_check.go:172]: we can move this to the top of the `Run` function for clarity

**Review by harveyxia — APPROVED** [review]

**Pull request merged**